contracts/staking/test/utils/api_wrapper.ts

```diff
@@ -0,0 +1,254 @@
+import { ERC20ProxyContract } from '@0x/contracts-asset-proxy';
+import { artifacts as erc20Artifacts, DummyERC20TokenContract } from '@0x/contracts-erc20';
+import { BlockchainTestsEnvironment, constants as testUtilsConstants, txDefaults } from '@0x/contracts-test-utils';
+import { SignatureType } from '@0x/types';
+import { BigNumber, logUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { ContractAbi, ContractArtifact, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import {
+    artifacts,
+    EthVaultContract,
+    StakingContract,
+    StakingPoolRewardVaultContract,
+    StakingProxyContract,
+    ZrxVaultContract,
+} from '../../src';
+
+import { ApprovalFactory } from './approval_factory';
+import { constants } from './constants';
+import { SignedStakingPoolApproval, StakeInfo } from './types';
+
+export class StakingApiWrapper {
+    public stakingProxyContract: StakingContract; // Wrapped as StakingContract to borrow API
+    public stakingContract: StakingContract; // The real Staking.sol contract
+    public zrxVaultContract: ZrxVaultContract;
+    public ethVaultContract: EthVaultContract;
+    public rewardVaultContract: StakingPoolRewardVaultContract;
+    public zrxTokenContract: DummyERC20TokenContract;
+    public utils = {
```

Comment:
I like this!

Reply:
Any reason why these functions aren't just at the root level of the `StakingApiWrapper` class? Otherwise the root level object seems less of a "wrapper" and more of just data fields.

Reply:
my rationale was that it looks a bit weird to have these functions at the same level as the contracts, i.e. I think `stakingApiWrapper.createStakingPoolAsync(...)` would look kind of weird alongside stuff like `stakingApiWrapper.stakingProxyContract.getStakingPoolIdOfMaker.callAsync(this._owner)`

Reply:
idk, doesn't look weird to me, but I'm possibly in the minority